src/resources/js/aggregator-fields.js

```diff
@@ -20,149 +27,321 @@ tribe_aggregator.fields = {
+				$( obj.selector.preview_button ).prop( 'disabled', false );
+			}
+		} );
+	};
+
+	/**
+	 * Initializes the datatable
+	 *
+	 * @param array data Array of events to display in the table
+	 */
+	obj.init_datatable = function( data ) {
+		var display_checkboxes = false;
 
-				var $import_type = $( '[id$="import_type"]:visible' );
-				if ( ! $import_type.length || 'manual' === $( '#' + $import_type.first().attr( 'id' ).replace( 's2id_', '' ) ).val() ) {
-					template_data.display_checkboxes = true;
+		var $import_type = $( '[id$="import_type"]:visible' );
+
+		if ( ! $import_type.length || 'manual' === $( '#' + $import_type.first().attr( 'id' ).replace( 's2id_', '' ) ).val() ) {
+			display_checkboxes = true;
+		}
+
+		var $table = obj.$.preview_container.find( '.data-container table' );
+
+		var rows = [];
+		for ( var i in data ) {
+			var row = data[ i ];
+			row.checkbox = display_checkboxes ? '<input type="checkbox">' : '';
+			rows.push( row );
+		}
+
+		if ( display_checkboxes ) {
+			$table.addClass( 'display-checkboxes' );
+		} else {
+			$table.removeClass( 'display-checkboxes' );
+		}
+
+		obj.$.preview_container.addClass( 'show-data' );
+		$table.tribeDataTable( {
+			lengthMenu: [
+				[5, 10, 25, 50, -1],
+				[5, 10, 25, 50, tribe_l10n_datatables.pagination.all ]
+			],
+			order: [
+				[ 1, 'asc' ]
+			],
+			columnDefs: [
+				{
+					cellType: 'th',
+					className: 'check-column',
+					orderable: false,
+					targets: 0
 				}
+			],
+			columns: [
+				{ data: 'checkbox' },
+				{ data: 'start_date' },
+				{ data: 'end_date' },
+				{ data: 'title' }
+			],
+			data: rows
+		} );
 
-				$( '.tribe-ea-table-container' ).append( template( template_data ) );
-				$( '.tribe-ea-table-container table').tribeDataTable( {
-					lengthMenu: [
-						[5, 10, 25, 50, -1],
-						[5, 10, 25, 50, tribe_l10n_datatables.pagination.all ]
-					],
-					order: [
-						[ 1, 'asc' ]
-					],
-					columnDefs: [
-						{
-							orderable: false,
-							targets: 0
-						}
-					]
-				} );
+		$table
+			.on( 'select.dt'  , obj.events.twiddle_finalize_button_text )
+			.on( 'deselect.dt', obj.events.twiddle_finalize_button_text );
 
-				var $preview_container = $( '.tribe-preview-container' );
-				$preview_container.removeClass( 'tribe-fetching' ).addClass( 'tribe-fetched' );
+		var text = obj.l10n.import_all.replace( '%d', rows.length );
+		$( obj.selector.finalize_button ).html( text );
+	};
+
+	/**
+	 * Displays a fetch error
+	 */
+	obj.display_fetch_error = function( message ) {
+		obj.$.preview_container.removeClass( 'tribe-fetching' ).addClass( 'tribe-fetch-error' );
+		obj.display_error( $( '.tribe-fetch-error-message' ), message );
+		$( obj.selector.preview_button ).prop( 'disabled', false );
+	};
+
+	/**
+	 * displays an error to a container on the page
+	 */
+	obj.display_error = function( $container, message ) {
+		$container.prepend(
+			[
```

Comment:
Not against this, just curious - why not concatenate string literals instead of placing them in an array and _then_ concatenating - an aesthetic thing?